include/nbl/video/SPhysicalDeviceFeatures.h

```diff
@@ -0,0 +1,508 @@
+    //VkBool32           shaderSharedInt64Atomics;
+    //VkBool32           shaderFloat16;
+    //VkBool32           shaderInt8;
+    //VkBool32           descriptorIndexing;
+    //VkBool32           shaderInputAttachmentArrayDynamicIndexing;
+    //VkBool32           shaderUniformTexelBufferArrayDynamicIndexing;
+    //VkBool32           shaderStorageTexelBufferArrayDynamicIndexing;
+    //VkBool32           shaderUniformBufferArrayNonUniformIndexing;
+    //VkBool32           shaderSampledImageArrayNonUniformIndexing;
+    //VkBool32           shaderStorageBufferArrayNonUniformIndexing;
+    //VkBool32           shaderStorageImageArrayNonUniformIndexing;
+    //VkBool32           shaderInputAttachmentArrayNonUniformIndexing;
+    //VkBool32           shaderUniformTexelBufferArrayNonUniformIndexing;
+    //VkBool32           shaderStorageTexelBufferArrayNonUniformIndexing;
+    //VkBool32           descriptorBindingUniformBufferUpdateAfterBind;
+    //VkBool32           descriptorBindingSampledImageUpdateAfterBind;
+    //VkBool32           descriptorBindingStorageImageUpdateAfterBind;
+    //VkBool32           descriptorBindingStorageBufferUpdateAfterBind;
+    //VkBool32           descriptorBindingUniformTexelBufferUpdateAfterBind;
+    //VkBool32           descriptorBindingStorageTexelBufferUpdateAfterBind;
+    //VkBool32           descriptorBindingUpdateUnusedWhilePending;
+    //VkBool32           descriptorBindingPartiallyBound;
+    //VkBool32           descriptorBindingVariableDescriptorCount;
+    //VkBool32           runtimeDescriptorArray;
+    //VkBool32           samplerFilterMinmax;
+    //VkBool32           scalarBlockLayout;
+    //VkBool32           imagelessFramebuffer;
+    //VkBool32           uniformBufferStandardLayout;
+    //VkBool32           shaderSubgroupExtendedTypes;
+    //VkBool32           separateDepthStencilLayouts;
+    //VkBool32           hostQueryReset;
+    //VkBool32           timelineSemaphore;
+    bool bufferDeviceAddress = false;
+    //VkBool32           bufferDeviceAddressCaptureReplay;
+    //VkBool32           bufferDeviceAddressMultiDevice;
+    //VkBool32           vulkanMemoryModel;
+    //VkBool32           vulkanMemoryModelDeviceScope;
+    //VkBool32           vulkanMemoryModelAvailabilityVisibilityChains;
+    //VkBool32           shaderOutputViewportIndex;
+    //VkBool32           shaderOutputLayer;
+    //VkBool32           subgroupBroadcastDynamicId;
+
+    /* Vulkan 1.3 Core */
+    //VkBool32           robustImageAccess;
+    //VkBool32           inlineUniformBlock;
+    //VkBool32           descriptorBindingInlineUniformBlockUpdateAfterBind;
+    //VkBool32           pipelineCreationCacheControl;
+    //VkBool32           privateData;
+    //VkBool32           shaderDemoteToHelperInvocation;
+    //VkBool32           shaderTerminateInvocation;
+    //VkBool32           subgroupSizeControl;
+    //VkBool32           computeFullSubgroups;
+    //VkBool32           synchronization2;
```

Comment:
Eventually we will require `KHR_synchronization2` and always enable it, because:

> It also adds new pipeline stage and access flag types that extend into the 64-bit range, as we have run out within the 32-bit range.

but probably not now cause our Barrier API would need to change :(


And the coverage of support of `KHR_synchronization2` is really poor on Vk 1.1 and 1.2 devices (Although I have no way to filter report duplicates which have had drivers upgraded to 1.3 or finally got the extension)

Reply:
make an issue about it and remark the only blocker is the adoption in 1.1 and 1.2 implementations which have not upgraded to 1.3 yet

Reply:
create the issue and update the comment to have a link to the issue